bundles/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/IhcGenericBindingProvider.java

```diff
@@ -169,10 +276,25 @@ public int getRefreshInterval(String itemName) {
 	}
 
 	@Override
-	public boolean isOutBindingOnly(String itemName) {
+	public boolean isOutBinding(String itemName, int resourceId) {
 		IhcBindingConfig config = (IhcBindingConfig) bindingConfigs
 				.get(itemName);
-		return config != null ? config.outBindingOnly : null;
+		
+		boolean isOutBinding = true;
+		
+		if(config.resourceId == resourceId) {
+			isOutBinding = false;
+		}
+		
+		
+		return isOutBinding;
+	}
+	
+	@Override
+	public boolean hasInBinding(String itemName) {
+		IhcBindingConfig config = (IhcBindingConfig) bindingConfigs.get(itemName);
+		
+		return config != null ? config.resourceId > 0 : null;
```

Comment:
There exists two classes: IhcOutCommandConfig (Configuration for an out configuration, this is used to configure resources for outgoing "commands").

In addition there is the IhcBindingConfig class. This class groups all configurations available to an item. The resourceId in this class is used for the InBinding and is by default 0.

When you look at the line 118 you will see that the resourceId is set at the IhcOutCommandConfig and not on the IhcBindingConfig and therefore the resourceId responsible for the In-Binding remains zero.